### PR TITLE
Add 342 release notes

### PR DESCRIPTION
**Pull request by martint**
Closes https://github.com/prestosql/presto/issues/5111

### CODE REVIEW

File: presto-docs/src/main/sphinx/release/release-342.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# Release 342 (XX Sep 2020)
+
+## General Changes
+
+* Improve performance of queries that use decimal type. ({issue}`4886`)
+* Improve query performance by adding support for dynamic filtering and dynamic
```
Comment (losipiuk) [inline presto-docs/src/main/sphinx/release/release-342.md:6]: Maybe w should describe it in terms of SQL syntax, not internal operator. 
Sth like:
```
Improve performance of queries which use ``expression IN SELECT ...`` filters. This is achieved via additional support for dynamic filtering an dynamic partition pruning.
```

File: presto-docs/src/main/sphinx/release/release-342.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  partition pruning to semi-join relational operator. ({issue}`5017`)
+* Improve dynamic partition pruning and query performance by reducing latency of dynamic filters collection. ({issue}`4988`)
```
Comment (losipiuk) [inline presto-docs/src/main/sphinx/release/release-342.md:8]:
```
Reduce dynamic partition pruning overhead?
```

File: presto-docs/src/main/sphinx/release/release-342.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+* Improve join performance when cost-based optimizer has missing or inaccurate stats. ({issue}`5141`)
+* Enable connectors to wait for dynamic filters derived from replicated join before generating splits. ({issue}`4685`)
```
Comment (losipiuk) [inline presto-docs/src/main/sphinx/release/release-342.md:10]: I think this one requires explanation what it actually means for the user.

File: presto-docs/src/main/sphinx/release/release-342.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+## Atop Connector Changes
+
+* Fix incorrect query results when query contains predicates on `start_time` or `end_time` column. ({issue}`5125`)
+
+## Hive Connector Changes
+
+* Allow specifying what happens if data is inserted into existing Hive partition. 
+  This can be done using `hive.insert-existing-partitions-behavior` config property. ({issue}`4999`)
+* Add property (`hive.dynamic-filtering-probe-blocking-timeout`) for delaying table scans
+  until dynamic partition pruning can be performed more efficiently. ({issue}`4991`)
```
Comment (losipiuk) [inline presto-docs/src/main/sphinx/release/release-342.md:22]: It feels to me that we are using dynamic partition pruning and dynamic filtering interchangeably without consequence. I think for changes which relate to both we should stick to dynamic filtering, as this is more general feature and only use dynamic partition pruning for specific cases.

File: presto-docs/src/main/sphinx/release/release-342.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+* Disable matching the existing user and group of the table or partition when creating new files on HDFS.
```
Comment (losipiuk) [inline presto-docs/src/main/sphinx/release/release-342.md:23]: I do not understand what `matching` mean in this context (or rather I would not know if I did not read the PR).

File: presto-docs/src/main/sphinx/release/release-342.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    
+## Kafka Connector Changes
+
+* Expose message headers as a `_headers` column of `map(VARCHAR, array(VARBINARY))` type. ({issue}`4462`)
+* Add write support for `TIME`, `TIME WITH TIME ZONE`, `TIMESTAMP` and `TIMESTAMP WITH TIME ZONE` 
+  for Kafka connector when JSON encoder is in use. ({issue}`4743`)
+  
+## MySQL Connector Changes
+
+* Improve performance of `INSERT` statement when MySQL instance isn't running with GTID mode. ({issue}`4995`)
```
Comment (losipiuk) [inline presto-docs/src/main/sphinx/release/release-342.md:35]:
```suggestion
* Improve performance of `INSERT` statement when MySQL instance isn't running in GTID mode. ({issue}`4995`)
```

File: presto-docs/src/main/sphinx/release/release-342.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
+# Release 342 (XX Sep 2020)
+
+## General Changes
+
+* Improve performance of queries that use decimal type. ({issue}`4886`)
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-342.md:5]: "the decimal"

also "decimal" or "`decimal`"?

File: presto-docs/src/main/sphinx/release/release-342.md (hunk 2 above)
Comment (findepi) [reply presto-docs/src/main/sphinx/release/release-342.md:8]: i think "latency" is an important aspect 

but, this could be expressed more from end-user perspective

"Improve performance and latency of queries leveraging dynamic partition pruning with dynamic filters."

(i know it's repetitive but i think it's important to stick "dynamic filters" here, as this is the recognized feature name)

File: presto-docs/src/main/sphinx/release/release-342.md [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+* Improve of queries involving `IN` with subqueries by extending support for dynamic
+  filtering and dynamic partition pruning. ({issue}`5017`)
+* Improve dynamic partition pruning and query performance by reducing latency of dynamic filters collection. ({issue}`4988`)
+* Improve join performance when cost-based optimizer has missing or inaccurate stats. ({issue}`5141`)
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-342.md:9]: "inaccurate" or "incomplete"?

also, it's a table who has missing stats, not the cost-based optimizer, so:

"Improve performance of queries joining tables with missing or incomplete column statistics when the cost-based optimizer is enabled (which is the default)."

(doesn't sound great....)

File: presto-docs/src/main/sphinx/release/release-342.md (hunk 3 above)
Comment (findepi) [reply presto-docs/src/main/sphinx/release/release-342.md:10]: is this a SPI change? @sopel39

File: presto-docs/src/main/sphinx/release/release-342.md [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+* Fix query failure when lambda expression references a table column containing a dot. ({issue}`5087`)
+
+## Atop Connector Changes
+
+* Fix incorrect query results when query contains predicates on `start_time` or `end_time` column. ({issue}`5125`)
+
+## Hive Connector Changes
+
+* Allow specifying what happens if data is inserted into existing Hive partition. 
+  This can be done using `hive.insert-existing-partitions-behavior` config property. ({issue}`4999`)
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-342.md:20]: "Allow specifying" is a bit of overstatement, as this was existing feature that we made **more configurable.**

BTW @losipiuk @kokosing why did we not deprecate the `hive.immutable-partitions`?

File: presto-docs/src/main/sphinx/release/release-342.md [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+* Add property (`hive.dynamic-filtering-probe-blocking-timeout`) for delaying table scans
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-342.md:21]: is there mention this is also configurable with session `dynamic_filtering_probe_blocking_timeout`

File: presto-docs/src/main/sphinx/release/release-342.md (hunk 5 above)
Comment (findepi) [reply presto-docs/src/main/sphinx/release/release-342.md:23]: @losipiuk the wording is based on how we announced the change in 341:
```
(341)
* Match the existing user and group of the table or partition when creating new files on HDFS. ({issue}`4414`)
```

i am not saying this is readable though

File: presto-docs/src/main/sphinx/release/release-342.md [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+  until dynamic partition pruning can be performed more efficiently. ({issue}`4991`)
+* Disable matching the existing user and group of the table or partition when creating new files on HDFS.
+  The functionality was added in 341 and is now disabled by default. You can enable it with `hive.fs.new-file-inherit-ownership`
+  configuration property. ({issue}`5187`)
+    
+## Kafka Connector Changes
+
+* Expose message headers as a `_headers` column of `map(VARCHAR, array(VARBINARY))` type. ({issue}`4462`)
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-342.md:29]: lowercase VARCHAR and VARBINARY

File: presto-docs/src/main/sphinx/release/release-342.md [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+* Add write support for `TIME`, `TIME WITH TIME ZONE`, `TIMESTAMP` and `TIMESTAMP WITH TIME ZONE`
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-342.md:30]: lowercase the types

File: presto-docs/src/main/sphinx/release/release-342.md [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+  until dynamic partition pruning can be performed more efficiently. ({issue}`4991`)
+* Disable matching the existing user and group of the table or partition when creating new files on HDFS.
+  The functionality was added in 341 and is now disabled by default. You can enable it with `hive.fs.new-file-inherit-ownership`
+  configuration property. ({issue}`5187`)
+    
+## Kafka Connector Changes
+
+* Expose message headers as a `_headers` column of `map(VARCHAR, array(VARBINARY))` type. ({issue}`4462`)
+* Add write support for `TIME`, `TIME WITH TIME ZONE`, `TIMESTAMP` and `TIMESTAMP WITH TIME ZONE` 
+  for Kafka connector when JSON encoder is in use. ({issue}`4743`)
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-342.md:31]: "when JSON encoder is in use" -> "when using JSON encoder"

File: presto-docs/src/main/sphinx/release/release-342.md [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+  
+## MySQL Connector Changes
+
+* Improve performance of `INSERT` statement when MySQL instance isn't running in GTID mode. ({issue}`4995`)
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-342.md:35]: isn't -> is not 
(or even present simple)

or maybe " .... (unless MySQL instance runs in GTID mode)" -- assuming GTID is an unusual configuration?

File: presto-docs/src/main/sphinx/release/release-342.md [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+
+## PostgreSQL Connector Changes
+
+* Extend type mapping to support variadic `TIMESTAMP` and `TIMESTAMP WITH ZONE` types. ({issue}`5124`, {issue}`5105`)
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-342.md:39]: lowercase the types

would s/variadic/parametric/ be acceptable (in the interest of making this easier to understand)?

File: presto-docs/src/main/sphinx/release/release-342.md [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+
+## SQL Server Connector Changes
+
+* Fix failure when inserting `NULL` to `VARBINARY` column. ({issue}`4846`)
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-342.md:43]: lowercase varbinary

File: presto-docs/src/main/sphinx/release/release-342.md [hunk 17]
```diff
@@ -0,0 +1,52 @@
...
+* Improve performance of aggregation queries by computing aggregations within SQL Server database.
+  Currently, the following aggregate functions are eligible for pushdown:
+  ``count``,  ``min``, ``max``, ``sum`` and ``avg``. ({issue}`4139`)
+
+## SPI Changes
+
+* Make dynamic filter futures resilient to cancellation. ({issue}`5099`)
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-342.md:50]: This is a change in engine's behavior, but is it really a SPI change?
@sopel39

File: presto-docs/src/main/sphinx/release/release-342.md [hunk 18]
```diff
@@ -0,0 +1,52 @@
...
+* Add `DynamicFilter#isAwaitable` method that returns whether dynamic filter is not complete and can be
+  awaited for via future. ({issue}`5043`)
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-342.md:52]: remove "via future" (or add information what future this is ref'ing to)

File: presto-docs/src/main/sphinx/release/release-342.md [hunk 19]
```diff
@@ -0,0 +1,52 @@
+# Release 342 (XX Sep 2020)
+
+## General Changes
+
+* Improve performance of queries that use decimal type. ({issue}`4886`)
+* Improve of queries involving `IN` with subqueries by extending support for dynamic
```
Comment (sopel39) [inline presto-docs/src/main/sphinx/release/release-342.md:6]: `Improve performance or`

File: presto-docs/src/main/sphinx/release/release-342.md (hunk 8 above)
Comment (sopel39) [reply presto-docs/src/main/sphinx/release/release-342.md:9]: > also, it's a table who has missing stats, not the cost-based optimizer, so:

technically, intermediate cbo-stats stats can be inaccurate too. I'm fine with your version too

File: presto-docs/src/main/sphinx/release/release-342.md (hunk 3 above)
Comment (sopel39) [reply presto-docs/src/main/sphinx/release/release-342.md:10]: this should be in SPI notes

File: presto-docs/src/main/sphinx/release/release-342.md (hunk 4 above)
Comment (sopel39) [reply presto-docs/src/main/sphinx/release/release-342.md:22]: makes sense, let's stick to `dynamic filtering`

File: presto-docs/src/main/sphinx/release/release-342.md (hunk 17 above)
Comment (sopel39) [reply presto-docs/src/main/sphinx/release/release-342.md:50]: It would be too technical for general change. It's mostly relevant to SPI

File: presto-docs/src/main/sphinx/release/release-342.md (hunk 18 above)
Comment (sopel39) [reply presto-docs/src/main/sphinx/release/release-342.md:52]:
```
Add `DynamicFilter#isAwaitable` method that returns whether dynamic filter is not complete and can be
  awaited for using `DynamicFilter#isBlocked`. ({issue}`5043`)
```
?

File: presto-docs/src/main/sphinx/release/release-342.md (hunk 17 above)
Comment (findepi) [reply presto-docs/src/main/sphinx/release/release-342.md:50]: Is it relevant to a connector's implementor?
if not, maybe we just remove it from here? @sopel39 ?

Comment (sopel39) [reply presto-docs/src/main/sphinx/release/release-342.md:50]: I haven't seen people complaining :) about this issue. OK to remove it

File: presto-docs/src/main/sphinx/release/release-342.md (hunk 9 above)
Comment (dain) [reply presto-docs/src/main/sphinx/release/release-342.md:23]: That flag allows the admin disable modification of existing partitions in a way that can not be overridden by end users.

File: presto-docs/src/main/sphinx/release/release-342.md (hunk 15 above)
Comment (dain) [reply presto-docs/src/main/sphinx/release/release-342.md:49]: I would say "with precision" instead of a technical term like variadic

File: presto-docs/src/main/sphinx/release/release-342.md (hunk 14 above)
Comment (ebyhr) [reply presto-docs/src/main/sphinx/release/release-342.md:45]: I prefer latter as GTID is special setting.

File: presto-docs/src/main/sphinx/release/release-342.md [hunk 20]
```diff
@@ -0,0 +1,62 @@
+# Release 342 (XX Sep 2020)
```
Comment (losipiuk) [inline presto-docs/src/main/sphinx/release/release-342.md:1]: TODO

File: presto-docs/src/main/sphinx/release/release-342.md (hunk 14 above)
Comment (losipiuk) [reply presto-docs/src/main/sphinx/release/release-342.md:45]: According to documentation guidelines `isn't` is preferred over `is not` (https://developers.google.com/style/contractions#negation-contractions)

Comment (losipiuk) [reply presto-docs/src/main/sphinx/release/release-342.md:45]: Yet I also prefer the version with unless

File: presto-docs/src/main/sphinx/release/release-342.md (hunk 15 above)
Comment (losipiuk) [reply presto-docs/src/main/sphinx/release/release-342.md:49]: TIMESTAMP WITH ZONE -> TIMESTAMP WITH TIME ZONE

File: presto-docs/src/main/sphinx/release/release-342.md (hunk 12 above)
Comment (losipiuk) [reply presto-docs/src/main/sphinx/release/release-342.md:34]: @findepi We are not super consistent in the documentation but most of the time we use uppercase. Was there a decision made to switch to lowercase?
cc: @mosabua

Comment (findepi) [reply presto-docs/src/main/sphinx/release/release-342.md:34]: i don't think there was "a switch". i know we [tried to] use[d] lowercase for presto type names in release notes before.

maybe we just need to re-decide _and_ write the decision down somewhere?

cc @electrum

File: presto-docs/src/main/sphinx/release/release-342.md (hunk 14 above)
Comment (martint) [reply presto-docs/src/main/sphinx/release/release-342.md:45]: @findepi, are you suggesting "Improve performance of `INSERT` statement unless MySQL instance runs in GTID mode"? That's doesn't read well.

Maybe something like this:

"Improve performance of `INSERT` queries when GTID mode is disabled in MySQL".

File: presto-docs/src/main/sphinx/release/release-342.md (hunk 15 above)
Comment (martint) [reply presto-docs/src/main/sphinx/release/release-342.md:49]: Yes, and "variadic" is the wrong term. It means "variable number of arguments". It should be "variable precision"

File: presto-docs/src/main/sphinx/release/release-342.md [hunk 21]
```diff
@@ -0,0 +1,77 @@
...
+
+* Expose message headers as a `_headers` column of `map(varchar, array(varbinary))` type. ({issue}`4462`)
+* Add write support for `TIME`, `TIME WITH TIME ZONE`, `TIMESTAMP` and `TIMESTAMP WITH TIME ZONE` 
+  for Kafka connector when using the JSON encoder. ({issue}`4743`)
+* In JSON decoder drop decoding support for temporal types for nonsensical combinations of input-format-type/data-type.
+  Following combination are no longer supported:
+  - `rfc2822`:  `DATE`, `TIME`, `TIME WITH TIME ZONE`
+  - `milliseconds-since-epoch`: `TIME WITH TIME ZONE`, `TIMESTAMP WITH TIME ZONE`
+  - `seconds-since-epoch`: `TIME WITH TIME ZONE`, `TIMESTAMP WITH TIME ZONE`
+  ({issue}`4743`)
```
Comment (martint) [inline presto-docs/src/main/sphinx/release/release-342.md:56]: This doesn't format well. The issue link is attached to the last item in this list.

File: presto-docs/src/main/sphinx/release/release-342.md [hunk 22]
```diff
@@ -0,0 +1,77 @@
+# Release 342 (XX Sep 2020)
+
+## General Changes
+
+* Add {func}`from_iso8601_timestamp_nanos` function. ({issue}`5048`)
+* Improve performance of queries that use the `DECIMAL` type. ({issue}`4886`)
+* Improve performance of queries involving `IN` with subqueries by extending support for dynamic
+  filtering and dynamic partition pruning. ({issue}`5017`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-342.md:8]: What does "partition pruning" mean in the context of the engine?

File: presto-docs/src/main/sphinx/release/release-342.md [hunk 23]
```diff
@@ -0,0 +1,77 @@
...
+* Improve performance and latency of queries leveraging dynamic partition pruning with dynamic filters. ({issue}`4988`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-342.md:9]: Same question as above

File: presto-docs/src/main/sphinx/release/release-342.md [hunk 24]
```diff
@@ -0,0 +1,77 @@
...
+* Improve performance of queries joining tables with missing or incomplete column statistics when
+  the cost-based optimizer is enabled (which is the default). ({issue}`5141`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-342.md:11]: We don't use the phrase "cost-based optimizer" in the documentation (since it's not correct)

> when cost based optimization is enabled

File: presto-docs/src/main/sphinx/release/release-342.md [hunk 25]
```diff
@@ -0,0 +1,77 @@
...
+* Reduce latency for queries where broadcast join is used and broadcasted table is large. ({issue}`5237`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-342.md:12]: > Reduce latency for queries that perform a broadcast join of a large table.

File: presto-docs/src/main/sphinx/release/release-342.md [hunk 26]
```diff
@@ -0,0 +1,77 @@
...
+
+## Elasticsearch Connector Changes
+
+* Allow reading boolean values stored as strings. ({issue}`5269`)
+
+## Hive Connector Changes
+
+* Add support for S3 encrypted files. ({issue}`2536`)
+* Add support for ABFS OAuth authentication. ({issue}`5052`)
+* Support reading timestamp with microsecond or nanosecond precision. This can be enabled with `hive.timestamp-precision`
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-342.md:27]: "with the xxx configuration property"

File: presto-docs/src/main/sphinx/release/release-342.md [hunk 27]
```diff
@@ -0,0 +1,77 @@
...
+  connector configuration property. ({issue}`4953`)
+* Allow specifying what happens if data is inserted into existing Hive partition.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-342.md:29]: > Allow overwrite on insert by default using the xxx configuration property.

File: presto-docs/src/main/sphinx/release/release-342.md [hunk 28]
```diff
@@ -0,0 +1,77 @@
...
+
+## Hive Connector Changes
+
+* Add support for S3 encrypted files. ({issue}`2536`)
+* Add support for ABFS OAuth authentication. ({issue}`5052`)
+* Support reading timestamp with microsecond or nanosecond precision. This can be enabled with `hive.timestamp-precision`
+  connector configuration property. ({issue}`4953`)
+* Allow specifying what happens if data is inserted into existing Hive partition. 
+  This can be done using `hive.insert-existing-partitions-behavior` config property. ({issue}`4999`)
+* Add property (`hive.dynamic-filtering-probe-blocking-timeout`) for delaying table scans
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-342.md:31]: > Allow delaying table scans until dynamic filtering can be performed more efficiently. This can be enabled using the xxx configuration property or the xxx session property.

File: presto-docs/src/main/sphinx/release/release-342.md [hunk 29]
```diff
@@ -0,0 +1,77 @@
...
+  until dynamic filtering can be performed more efficiently. This can also be set via the `dynamic_filtering_probe_blocking_timeout`
+  session property. ({issue}`4991`)
+* Disable matching the existing user and group of the table or partition when creating new files on HDFS.
+  The functionality was added in 341 and is now disabled by default. You can enable it with `hive.fs.new-file-inherit-ownership`
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-342.md:35]: > It can be enabled using the xxx configuration property.

File: presto-docs/src/main/sphinx/release/release-342.md [hunk 30]
```diff
@@ -0,0 +1,77 @@
...
+  configuration property. ({issue}`5187`)
+* Improve performance of reading small file in RCFile format. ({issue}`2536`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-342.md:37]: > Improve performance when reading small files in `RCTEXT` or `RCBINARY` format.

File: presto-docs/src/main/sphinx/release/release-342.md (hunk 14 above)
Comment (electrum) [reply presto-docs/src/main/sphinx/release/release-342.md:60]: Contractions should generally only be used in friendly, informal writing. That's not how we currently write most of the documentation, including release notes.

File: presto-docs/src/main/sphinx/release/release-342.md [hunk 31]
```diff
@@ -0,0 +1,77 @@
...
+# Iceberg Connector Changes
+
+* Fix partition transforms for temporal columns for dates before 1970. ({issue}`5273`)
+
+## Kafka Connector Changes
+
+* Expose message headers as a `_headers` column of `map(varchar, array(varbinary))` type. ({issue}`4462`)
+* Add write support for `TIME`, `TIME WITH TIME ZONE`, `TIMESTAMP` and `TIMESTAMP WITH TIME ZONE` 
+  for Kafka connector when using the JSON encoder. ({issue}`4743`)
+* In JSON decoder drop decoding support for temporal types for nonsensical combinations of input-format-type/data-type.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-342.md:51]: > Remove JSON decoder support for nonsensical combinations of input-format-type / data-type. The following combinations are no longer supported:

File: presto-docs/src/main/sphinx/release/release-342.md (hunk 21 above)
Comment (electrum) [reply presto-docs/src/main/sphinx/release/release-342.md:56]: Maybe put it at the after the `:` before the list

File: presto-docs/src/main/sphinx/release/release-342.md (hunk 14 above)
Comment (electrum) [reply presto-docs/src/main/sphinx/release/release-342.md:60]: I like @martint's phrasing

File: presto-docs/src/main/sphinx/release/release-342.md [hunk 32]
```diff
@@ -0,0 +1,77 @@
...
+  - `seconds-since-epoch`: `TIME WITH TIME ZONE`, `TIMESTAMP WITH TIME ZONE`
+  ({issue}`4743`)
+  
+## MySQL Connector Changes
+
+* Improve performance of `INSERT` statement when MySQL instance isn't running in GTID mode. ({issue}`4995`)
+
+## PostgreSQL Connector Changes
+
+* Extend type mapping to support variable-precision `TIMESTAMP` and `TIMESTAMP WITH TIME ZONE` types. ({issue}`5124`, {issue}`5105`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-342.md:64]: I'm not sure what "extend type mapping" means. That sounds like an implementation detail. What is the user visible effect?

File: presto-docs/src/main/sphinx/release/release-342.md (hunk 16 above)
Comment (electrum) [reply presto-docs/src/main/sphinx/release/release-342.md:68]: We've been using uppercase in the others, so we should be consistent

File: presto-docs/src/main/sphinx/release/release-342.md [hunk 33]
```diff
@@ -0,0 +1,77 @@
...
+
+## SQL Server Connector Changes
+
+* Fix failure when inserting `NULL` to `VARBINARY` column. ({issue}`4846`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-342.md:68]: to -> into a

File: presto-docs/src/main/sphinx/release/release-342.md [hunk 34]
```diff
@@ -0,0 +1,77 @@
...
+* Improve performance of aggregation queries by computing aggregations within SQL Server database.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-342.md:69]: > Improve performance of aggregation queries by pushing aggregations into SQL Server.

File: presto-docs/src/main/sphinx/release/release-342.md [hunk 35]
```diff
@@ -0,0 +1,77 @@
...
+  Currently, the following aggregate functions are eligible for pushdown:
+  ``count``,  ``min``, ``max``, ``sum`` and ``avg``. ({issue}`4139`)
+
+## SPI Changes
+
+* Add `DynamicFilter#isAwaitable` method that returns whether dynamic filter is not complete and can be
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-342.md:75]: `DynamicFilter.isAwaitable()`

Comment (electrum) [reply presto-docs/src/main/sphinx/release/release-342.md:75]: Add `DynamicFilter.isAwaitable()` method that returns whether or not the dynamic filter is complete and can be awaited for using the `isBlocked()` method.

File: presto-docs/src/main/sphinx/release/release-342.md [hunk 36]
```diff
@@ -0,0 +1,77 @@
...
+  awaited for using `DynamicFilter#isBlocked`. ({issue}`5043`)
+* Enable connectors to wait for dynamic filters derived from replicated join before generating splits. ({issue}`4685`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-342.md:77]: join -> joins

File: presto-docs/src/main/sphinx/release/release-342.md (hunk 22 above)
Comment (sopel39) [reply presto-docs/src/main/sphinx/release/release-342.md:8]: Let's maybe use just `dynamic filtering` as an umbrella keyword.

File: presto-docs/src/main/sphinx/release/release-342.md (hunk 14 above)
Comment (findepi) [reply presto-docs/src/main/sphinx/release/release-342.md:60]: > "Improve performance of INSERT queries when GTID mode is disabled in MySQL".

@martint i read this as "if i disabled GTID mode my inserts will be faster now".

File: presto-docs/src/main/sphinx/release/release-342.md (hunk 24 above)
Comment (findepi) [reply presto-docs/src/main/sphinx/release/release-342.md:11]: I know it is not strictly correct. We used "cost-based optimizer" and "cost based optimizer" in the past, even though the correct term is "cost-based optimizations". -- It's because all the users understand "cost-based optimizer" and this is how they refer to this set of features ("lingua franca")

File: presto-docs/src/main/sphinx/release/release-342.md (hunk 32 above)
Comment (findepi) [reply presto-docs/src/main/sphinx/release/release-342.md:64]: @electrum end user -- timestamp(p) in PostgreSQL is now exposed as timestamp(p) in Presto (instead of timestamp(3) it used to be)

File: presto-docs/src/main/sphinx/release/release-342.md (hunk 34 above)
Comment (findepi) [reply presto-docs/src/main/sphinx/release/release-342.md:69]: @electrum i copied this exact wording from 337 RNs.

File: presto-docs/src/main/sphinx/release/release-342.md (hunk 16 above)
Comment (findepi) [reply presto-docs/src/main/sphinx/release/release-342.md:68]: @electrum @martint i see us using lowercase or uppercase depending which release it is
Also, it is very mixed today across docs.
Can we have an agreement & write it down?

File: presto-docs/src/main/sphinx/release/release-342.md (hunk 24 above)
Comment (mosabua) [reply presto-docs/src/main/sphinx/release/release-342.md:11]: We should just check what is grammatically correct and follows the style guide we use and then just update all the docs in one sweep

File: presto-docs/src/main/sphinx/release/release-342.md (hunk 32 above)
Comment (martint) [reply presto-docs/src/main/sphinx/release/release-342.md:64]: Maybe rephrase as this, then?

> Add support for variable-precision `TIMESTAMP` and `TIMESTAMP WITH TIME ZONE` types.

File: presto-docs/src/main/sphinx/release/release-342.md (hunk 24 above)
Comment (martint) [reply presto-docs/src/main/sphinx/release/release-342.md:11]: @findepi, on the other hand, we should break that habit. We shouldn't call the cost-based optimizer as if it were a separate thing. There's only one optimizer in Presto, with the ability to do cost-based optimizations for some plan shapes.

Comment (findepi) [reply presto-docs/src/main/sphinx/release/release-342.md]: > There's only one optimizer in Presto

that's true (assuming we don't count PlanOptimizer subclasses but rather PlanOptimizers class as a whole)

But that is also something users are not interested in.

Anyway, I am OK with the term "cost-based optimizations" (people will still read this as "cost-based optimizer" 😉)

File: presto-docs/src/main/sphinx/release/release-342.md (hunk 14 above)
Comment (martint) [reply presto-docs/src/main/sphinx/release/release-342.md:59]: Isn't that what the original text implied?

File: presto-docs/src/main/sphinx/release/release-342.md [hunk 37]
```diff
@@ -0,0 +1,76 @@
...
+## PostgreSQL Connector Changes
+
+* Add support for variable-precision TIMESTAMP and TIMESTAMP WITH TIME ZONE types. ({issue}`5124`, {issue}`5105`)
+
+## SQL Server Connector Changes
+
+* Fix failure when inserting `NULL` into a `VARBINARY` column. ({issue}`4846`)
+* Improve performance of aggregation queries by computing aggregations within SQL Server database.
+  Currently, the following aggregate functions are eligible for pushdown:
+  ``count``,  ``min``, ``max``, ``sum`` and ``avg``. ({issue}`4139`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-342.md:70]: Single backticks

**Review by electrum — APPROVED** [review]
